spec-examples/src/ntru_prime/ntru_prime.rs

```diff
@@ -0,0 +1,147 @@
+use hacspec::prelude::*;
+
+/// Struct to decide NtruVersion
+pub struct NtruVersion {
+    pub p: usize,
+    pub q: i128,
+    pub w: i128,
+    pub irr: Seq<i128>,
+}
+pub fn set_irr(p: usize) -> Seq<i128> {
+    let mut irr: Seq<i128> = Seq::new(p + 1);
+    irr[0] = -1 as i128;
+    irr[1] = -1 as i128;
+    irr[p] = 1 as i128;
+    irr
+}
+#[macro_export]
+macro_rules! ntru_v {
```

Comment:
So for this we need something better ...
Couldn't this be a function on `NtruVersion` (I'd rename that to `Ntru`)? Something like `fn new(version: NtruVersionEnum) -> Self`

Reply:
Ah right, it should be part of the test_ntru.rs probably